_plugins/jekyll-tabs-lib/jekyll-tabs-4scala.rb

```diff
@@ -97,25 +96,34 @@ def errorScalaVersionDefault(tab)
                         foundDefault = true
                     end
 
-                    isScalaTab = Tabs::ScalaVersions.include? tab.label
-
-                    if @is_scala_tabs
-                        if !isScalaTab
-                            raise errorNonScalaVersion(tab)
-                        elsif tab.defaultTab
-                            raise errorScalaVersionDefault(tab)
+                    def checkTab(tab, tabClass, expectedTabs, raiseIfMissingClass)
+                        isValid = expectedTabs.include? tab.label
+                        if @tab_class == tabClass
+                            if !isValid
+                                raise errorInvalidTab(tab, expectedTabs)
+                            elsif tab.defaultTab
+                                raise errorTabDefault(tab)
+                            end
+                        elsif raiseIfMissingClass and isValid
+                            raise errorTabWithoutClass(tab, tabClass)
                         end
-                    elsif !@is_scala_tabs and isScalaTab
-                        raise errorScalaVersionWithoutClass(tab)
                     end
+
+                    checkTab(tab, "tabs-scala-version", Tabs::ScalaVersions, true)
+                    checkTab(tab, "tabs-build-tool", Tabs::BuildTools, false)
                 end
 
-                if @is_scala_tabs and seenTabs != Tabs::ScalaVersions
-                    raise errorMissingScalaVersion()
+                def checkExhaustivity(seenTabs, tabClass, expectedTabs)
+                    if @tab_class == tabClass and seenTabs != expectedTabs
+                      raise errorMissingTab(expectedTabs)
+                    end
                 end
 
+                checkExhaustivity(seenTabs, "tabs-scala-version", Tabs::ScalaVersions)
+                checkExhaustivity(seenTabs, "tabs-build-tool", Tabs::BuildTools)
+
                 if !foundDefault and allTabs.length > 0
-                    if @is_scala_tabs
+                    if @tab_class == "tabs-scala-version" or @tab_class == "tabs-build-tool"
```

Comment:
currently we don't have validation of the order, but by the way tabs were arranged in this PR, this is going to display `Mill` first, shouldn't it be `Scala CLI`?